nuitka/build/static_src/CompiledFunctionType.c

```diff
@@ -25,6 +25,38 @@
 // Needed for offsetof
 #include <stddef.h>
 
+typedef struct {
+    PyObject_HEAD
+}  MetaObject;
+
+
+static PyObject *
+Check_instance(MetaObject *self, PyObject *Py_UNUSED(ignored))
+{
+    // PRINT_ITEM(self);
+    PRINT_NEW_LINE();
+    PRINT_STRING("Hello");
+}
+
+static PyMethodDef Meta_methods[] = {
+    {"type", (PyCFunction) Check_instance, METH_NOARGS,
+     "Return the name, combining the first and last name"
+    },
+    {NULL}  /* Sentinel */
+};
+
+static PyTypeObject MetaType = {
```

Comment:
I think you need to make "type" a base class, the crash is probably related to you "object" derived meta class being used a meta class, but not really being one. This is probably cruicial, but ought to lead to an exception, alAlso you didn't `PyType_Ready` it before you used it as a metaclass.

Compile with `--debug` and also check the return codes in an assert statement. We don't do that yet, but that was probably stupid of me.

Reply:
I have a doubt where can I `PyType_Ready` the MetaType. In the `_initCompiledFunctionType()` or should another `PyMODINIT_FUNC` function should be made.

Reply:
Just "ready" it before you "ready" the function type. Later we can think of sharing it with others, but now we are not going for re-usable to other compiled type yet.